README.md

```diff
@@ -60,28 +91,97 @@ can append the following section in your `Cargo.toml`.
 
-Instruments has the concept of 'templates', which describe sets of dtrace probes
-that can be enabled. `cargo-instruments` will use the "[Time
-Profiler][time profiler]", which collects CPU core and thread use.
+As usual, thanks to Clap, running `cargo instruments -h` prints the compact help.
 
-### examples
+    cargo-instruments 0.4.0
+    Profile a binary with Xcode Instruments.
+
+    By default, cargo-instruments will build your main binary.
+
+    USAGE:
+        cargo instruments [FLAGS] [OPTIONS] [ARGS]...
+
+    FLAGS:
+        -h, --help              Prints help information
+        -l, --list-templates    List available templates
+            --open              Open the generated .trace file after profiling
+            --release           Pass --release to cargo
+        -V, --version           Prints version information
+
+    OPTIONS:
+            --bench <NAME>                 Benchmark target to run
+            --bin <NAME>                   Binary to run
+            --example <NAME>               Example binary to run
+            --features <CARGO-FEATURES>    Features to pass to cargo
+        -t, --template <TEMPLATE>          Specify the instruments template to run
+            --time-limit <MILLIS>          Limit recording time to the specified value (in milliseconds)
+        -o, --output <PATH>                Output .trace file to the given path
+
+    ARGS:
+        <ARGS>...    Arguments passed to the target binary
+
+And `cargo instruments --help` provides more detail.
+
+### Templates
+
+Instruments has the concept of 'templates', which describe sets of dtrace
+probes that can be enabled. You can ask `cargo-instruments` to list available
+templates, including your custom ones (see help above). If you don't provide a
+template name, you will be prompted to choose one.
+
+Typically, the built-in templates are
+
+    built-in            abbrev
+    --------------------------
+    Activity Monitor
+    Allocations         (alloc)
+    Animation Hitches
+    App Launch
+    Core Data
+    Counters
+    Energy Log
+    File Activity       (io)
+    Game Performance
+    Leaks
+    Logging
+    Metal System Trace
+    Network
+    SceneKit
+    SwiftUI
+    System Trace        (sys)
+    Time Profiler       (time)
+    Zombies
+
+
+### Examples
 
 ```sh
 # View all args and options
 $ cargo instruments --help
 ```
 
+```sh
+# View all built-in and custom templates
+$ cargo instruments --list-templates
+```
+
 ```sh
 # profile the main binary with the Allocations template
 $ cargo instruments -t alloc
 ```
 
 ```sh
-# profile examples/my_example.rs, with the default template,
+# profile examples/my_example.rs, with the Allocations template,
 # for 10 seconds, and open the trace when finished
-$ cargo instruments --example my_example --limit 10000 --open
+$ cargo instruments -t Allocations --example my_example --time-limit 10000 --open
+```
+
+```sh
+# omitting to provide a template will display a `dialoguer` menu
```

Comment:
```suggestion
# omitting to provide a template will display a `dialoguer` menu
```

is 'dialogeur' a typo, or just a term I'm not familiar with?

Reply:
dialoguer is a crate for interactive menus: https://crates.io/crates/dialoguer
I will improve the wording.

Reply:
ah cool I haven't actually tried to run this yet, I should do that before I merge 😉